dev/TeachingTip/TeachingTip.cpp

```diff
@@ -830,6 +831,26 @@ void TeachingTip::OnIsOpenChanged()
                 }
             }
         }
+
+        m_acceleratorKeyActivatedRevoker = Dispatcher().AcceleratorKeyActivated(winrt::auto_revoke, { this, &TeachingTip::OnF6AcceleratorKeyClicked });
+
+        if (auto child = m_rootElement.get())
+        {
+            m_tipGettingFocusRevoker = AddRoutedEventHandler<RoutedEventType::GettingFocus>(
+                child,
+                [this](auto&& sender, auto&& args)
+                {
+                    m_hasFocusInSubtree = true;
```

Comment:
>m_hasFocusInSubtree = true; [](start = 20, length = 27)

Be aware that if anyone redirects focus after your GettingFocus handler runs, then focus may not really be in your sub-tree.  It may be unwise to try to track this state as it can be misleading/wrong sometimes, and just check if you have focus inside yourself every time you want to know, so you're guaranteed it's accurate each time.

Has more perf cost, but given how rarely this code will run that doesn't seem so bad?

Reply:
Ok, i'll go back to what Stephen had.